lib/lightning/collaboration/persistence_writer.ex

```diff
@@ -355,13 +357,19 @@ defmodule Lightning.Collaboration.PersistenceWriter do
       {:error, exception}
   end
 
-  defp merge_updates([update]), do: update
+  defp merge_updates([update], _document_name), do: update
```

Comment:
The fix for my issue #4154  overlaps with what you're doing here. I want to get rid of this function all together. If there's a single update **doesn't mean that**

1. we don't want to generate a delta between the current state of the document(`_document_name`)
2. also, properly encode the changes using `Yex.encode_state_as_update!`

with 2. I don't think that just returning the single update is the same as passing it through a new doc and encoding.